cassandra/connection.py

```diff
@@ -779,6 +779,7 @@ def __init__(self, host='127.0.0.1', port=9042, authenticator=None,
             self.ssl_options.update(self.endpoint.ssl_options or {})
         elif self.endpoint.ssl_options:
             self.ssl_options = self.endpoint.ssl_options
+        self._check_hostname = self.ssl_options.get('check_hostname', False)
```

Comment:
_check_hostname is no longer used for anything; see PYTHON-1331 for more details.  I kept the [logic](https://github.com/datastax/python-driver/blob/master/cassandra/connection.py#L935-L941) to call _validate_hostname() if this flag is set in order to allow custom connection subclasses to do additional validation if they wish but it's completely unnecessary.